src/cropareaest_funcs.py

```diff
@@ -0,0 +1,564 @@
+# Import libraries
+import os
+import pandas as pd
+import numpy as np
+import geopandas as gpd
+import rasterio as rio
+from rasterio.mask import mask
+import cartopy.io.shapereader as shpreader
+from shapely.geometry import box
+from sklearn.metrics import confusion_matrix
+
+
+def return_map_bounds_as_df():
+    "Read the bounding coordinates of the crop mask"
+    root_dir = os.getcwd()
+    prj_raster = [f for f in os.listdir(root_dir) if f.startswith("prj")][0]
+
+    with rio.open(prj_raster) as src:
+        bbx = src.bounds
+    geom = box(*bbx)
+    bbx_gdf = gpd.GeoDataFrame(geometry=[geom], crs=src.crs)
+    return bbx_gdf
+
+
+def load_ne(country_code, regions_of_interest):
+    """
+    Load the Natural Earth country and region shapefiles.
+    country_code: ISO 3166-1 alpha-3 country code
+    regions_of_interest: list of regions of interest
+    """
+
+    ne_shapefile = shpreader.natural_earth(
+        resolution="10m", category="cultural", name="admin_1_states_provinces"
+    )
+    ne_gdf = gpd.read_file(ne_shapefile)
+
+    if len(regions_of_interest) == 0:
+        # region_of_interest & country_code will be collected using dropdown
+        condition = ne_gdf["adm1_code"].str.startswith(country_code)
+        boundary = ne_gdf[condition].copy()
+        print("Entire country found!")
+
+    else:
+        # Check regions
+        available_regions = ne_gdf[ne_gdf["adm1_code"].str.startswith(country_code)][
+            "name"
+        ].tolist()
+        regions_not_found = [
+            region for region in regions_of_interest if region not in available_regions
+        ]
+
+        if len(regions_not_found) > 0:
+            condition = ne_gdf["adm1_code"].str.startswith(country_code)
+            boundary = None
+            print(
+                f"WARNING: {regions_not_found} was not found. Please select \
+                regions only seen in below plot."
+            )
+            ne_gdf[condition].plot(
+                column="name", legend=True, legend_kwds={"loc": "lower right"}, figsize=(10, 10)
+            )
+        else:
+            condition = ne_gdf["name"].isin(regions_of_interest)
+            boundary = ne_gdf[condition].copy()
+            print("All regions found!")
+    return boundary
```

Comment:
Ideally, once functions are moved out of a notebook they should get a unittest associated with them